cmd/jaeger/internal/extension/jaegerstorage/extension_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jaegerstorage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/extension"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+	"go.uber.org/zap"
+
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+const memstoreName = "memstore"
+
+type storageHost struct {
+	t                *testing.T
+	storageExtension component.Component
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	return map[component.ID]component.Component{
+		ID: host.storageExtension,
+	}
+}
+
+func (host storageHost) ReportFatalError(err error) {
+	host.t.Fatal(err)
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+func TestExtensionConfigError(t *testing.T) {
+	config := createDefaultConfig().(*Config)
+	err := config.Validate()
+	require.EqualError(t, err, fmt.Sprintf("%s: no storage type present in config", ID))
+}
+
+func TestStartStorageExtensionError(t *testing.T) {
+	ctx := context.Background()
+
+	storageExtension := makeStorageExtension(t, memstoreName)
+
+	host := componenttest.NewNopHost()
+	err := storageExtension.Start(ctx, host)
+	require.Error(t, err)
+	require.EqualError(t, err, fmt.Sprintf("duplicate memory storage name %s", memstoreName))
```

Comment:
Where is the error in this case?

Reply:
We are starting the storageExtension again. This will lead to an error. [Ref](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L65)

Reply:
but the errer message doesn't seem to be about extension starting.

Reply:
If you're testing specifically the dual entry condition, it would be better to test it explicitly, not via a side-effect of starting extension twice. I assume it's possible to create dual entry in the config if using a different type of storage and then giving it the same name.

Reply:
Currently only Memory type storage is supported. So I was unable to recreate this. Have changed the function name for now.